src/Signature/SignatureV4.php

```diff
@@ -14,21 +14,27 @@ class SignatureV4 implements SignatureInterface
 {
     use SignatureTrait;
     const ISO8601_BASIC = 'Ymd\THis\Z';
+    const UNSIGNED_PAYLOAD = 'UNSIGNED-PAYLOAD';
 
     /** @var string */
     private $service;
 
     /** @var string */
     private $region;
 
+    /** @var bool */
+    private $unsigned;
+
     /**
      * @param string $service Service name to use when signing
      * @param string $region  Region name to use when signing
+     * @param bool $unsigned  Flag used to set unsigned or signed payload
      */
-    public function __construct($service, $region)
+    public function __construct($service, $region, $unsigned = false)
```

Comment:
I like the idea of using an optional associative array for configuration options... Something like:

```
public function __construct($service, $region, array $options = [])
```

Where it accepts an "unsigned" key-value pair.